chutney/environment/src/main/java/com/chutneytesting/environment/domain/EnvironmentService.java

```diff
@@ -76,7 +80,15 @@ public Environment getEnvironment(String environmentName) throws EnvironmentNotF
     }
 
     public void deleteEnvironment(String environmentName) throws EnvironmentNotFoundException, CannotDeleteEnvironmentException {
+        List<String> environmentNames = environmentRepository.listNames();
+        if (environmentNames.size() == 1 && environmentNames.get(0).equals(environmentName)) {
```

Comment:
why not just if (environmentNames.size() == 1) ? I dont understant the second term